test/bench.cpp

```diff
@@ -4,87 +4,370 @@
+            local.push_back(inputs[i]);
+//            _stats.el.at(outputGate).push_back(std::to_string(getGate(inputs[i])));
+        _stats.el.push_back(local);
+        _stats.number_of_lines++;
+    }
+
+    virtual void on_assign(const std::string &input, const std::string &output) const override {
+        (void) input;
+        (void) output;
+        ++_stats.number_of_lines;
+    }
+
+    bench_statistics &_stats;
+    mutable std::vector<std::tuple<std::vector<std::string>, std::string, std::string>> gate_lines;
 }; /* bench_statistics_reader */
 
+//void secondCheck(bench_statistics &stats, int index_number) {
+//
+//    int index_numberLength = stats.el.at(index_number).size();
+//    int index_numberXCOORD = std::stoi(stats.el.at(index_number).at(index_numberLength - 3));
+//    for (int i = 2; i < stats.el.at(index_number).size() - 3; ++i) {
+//        int gate = getGateIndex(stats.el.at(index_number).at(i), stats);
+//        int length = stats.el.at(gate).size();
+//
+//        int gateXCoord = std::stoi(stats.el.at(gate).at(length - 3));
+//        if (gateXCoord > index_numberXCOORD) {
+//            //            при изменении x убрать значение y для предыдущего x!!!
+//            index_numberXCOORD = gateXCoord + 80;
+//            stats.el.at(index_number).at(index_numberLength - 3) = std::to_string(index_numberXCOORD);
+//        }
+//    }
+//    int y_coord = 200;
+//    if (!stats.x_coord.at(index_numberXCOORD).empty()) {
+//        int length = stats.x_coord.at(index_numberXCOORD).size();
+//        y_coord = stats.x_coord.at(index_numberXCOORD).at(length - 1) + 80;
+//        stats.x_coord.at(index_numberXCOORD).push_back(y_coord);
+//    }
+//    stats.x_coord.at(index_numberXCOORD).push_back(y_coord);
+//    stats.el.at(index_number).at(index_numberLength - 2) = std::to_string(y_coord);
+//}
+
 static void
 dump_statistics(FILE *f, const bench_statistics &st) {
-  fprintf(f, "inputs: %u, outputs: %u, num ddfs: %u, num lines: %u\n",
-      st.number_of_inputs,
-      st.number_of_outputs,
-      st.number_of_dffs,
-      st.number_of_lines);
+    fprintf(f, "inputs: %u, outputs: %u, num ddfs: %u, num lines: %u\n",
+            st.number_of_inputs,
+            st.number_of_outputs,
+            st.number_of_dffs,
+            st.number_of_lines);
+}
+
+void setElCoord(bench_statistics &stats, int index_number) {
+//function that specifies coordinate for all elements
+    int maxXCoord = -80;
+    int index_numberLength = stats.el.at(index_number).size();
+    if (stats.el.at(index_number).at(index_numberLength - 1) == "Coordinates set") return;
+    stats.el.at(index_number).push_back("Setting Coordinates...");
+
+    for (int i = 2; i < stats.el.at(index_number).size() - 1; ++i) {
+        int index_numberLength = stats.el.at(index_number).size();
+        if (stats.el.at(index_number).at(index_numberLength - 1) == "Coordinates set") break;
+        std::string gate = stats.el.at(index_number).at(i);
```

Comment:
`stats.el.at(index_number)` is accessed multiple times producing poorly readable code duplication, use a named reference instead, e.g. `std::vector<std::string> &data = stats.el[index_number];`